src/main/java/com/keniu/controllers/BookController.java

```diff
@@ -53,4 +54,22 @@ public BookDto getById(@PathVariable Long id) {
     public BookDto save(@RequestBody CreateBookRequestDto createBookRequestDto) {
         return bookService.save(createBookRequestDto);
     }
+
+    /**
+     * Updates the details of an existing book in the book shop.
+     *
+     * This method accepts a book ID and the new details of the book.
+     * It updates the existing book in the database
+     * with the provided information.
+     *
+     * @param id the unique identifier of the book to be updated
+     * @param createBookRequestDto
+     * a {@link CreateBookRequestDto} containing the new details of the book
+     * @return a {@link BookDto} representing the updated book with the new details
+     */
+    @PutMapping("/{id}")
+    public BookDto update(@PathVariable Long id,
+            @RequestBody CreateBookRequestDto createBookRequestDto) {
+        return bookService.update(id, createBookRequestDto);
+    }
```

Comment:
See task description 
```
Add missing endpoints and methods for BookService
DELETE /api/books/{id} (Delete a specific book)
```

Reply:
![Screenshot from 2024-11-09 16-20-55](https://github.com/user-attachments/assets/6af392fd-5721-4d1b-ba35-55e09d34b93e)
Look how it appears in the task. Here, delete is not part of the list, which is why I didn’t notice it. I think you can improve the task's readability by making delete part of the list, just like with the other endpoints. In any case, I’ve added this endpoint